### PR TITLE
Add caching to flags api and purge on deploy

### PR DESCRIPTION
**Pull request by aintgoin2goa**
Given that flags can only be changed by a code deploy there's no reason for it to not be cacheable.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,6 +16,4 @@ test:
 
 deploy:
 	next-build-tools deploy
-
-purge:
-	@curl --request PURGE --header "Fastly-Key: ${FASTLY_KEY}; Content-Accept: application/json" http://next.ft.com/__flags.json
+	 next-build-tools purge https://next.ft.com/__flags.json
```
Comment (matthew-andrews) [inline Makefile]: is https on a different cache key to http?

Comment (aintgoin2goa) [inline Makefile]: When I tested with http I got a 404 error for some reason

Comment (matthew-andrews) [inline Makefile]: Maybe not, this is the fastly gui that does it:-
screen shot 2015-04-09 at 07 50 29

Comment (matthew-andrews) [inline Makefile]: Seems they're the same keys!

```
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 183
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 312
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 191
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 316
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 188
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 325
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 323
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ next-build-tools purge http://next.ft.com/hashed-assets/engels/main-25ed0542.css
HTTP/1.1 404 Not Found
Date: Thu, 09 Apr 2015 07:06:09 GMT
Via: 1.1 localhost.localdomain
Content-Length: 0

HTTP/1.1 404 Not Found
Date: Thu, 09 Apr 2015 07:06:09 GMT
Via: 1.1 localhost.localdomain
Content-Length: 0



mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 350
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ next-build-tools purge https://next.ft.com/hashed-assets/engels/main-25ed0542.css
HTTP/1.1 404 Not Found
Date: Thu, 09 Apr 2015 07:06:32 GMT
Via: 1.1 localhost.localdomain
Content-Length: 0

HTTP/1.1 200 Connection established
Date: Thu, 09 Apr 2015 07:06:32 GMT
Age: 0
Via: 1.0 localhost.localdomain

HTTP/1.1 200 OK
Date: Thu, 09 Apr 2015 07:06:33 GMT
Server: Apache
Content-Type: application/json
Content-Length: 48
Accept-Ranges: bytes
X-Varnish: 2339182647
Via: 1.1 varnish

{"status": "ok", "id": "81-1426788950-4872477"}


mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 0
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl http://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 0
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 0
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $ curl https://next.ft.com/hashed-assets/engels/main-25ed0542.css -I -s | grep Age
Age: 0
mandrews@MATT ~/sandboxes/next-feature-flags-api (caching) $
```

**Comment by matthew-andrews** [comment]
:+1: looks very good to me.

**Comment by aintgoin2goa** [comment]
Yeah, I forgot about the proxy!  Ft has a craxy proxy which doesn't intercept https - which is why it worked for one and not the other - will merge!

**Comment by matthew-andrews** [comment]
you think it's that?  i was runnning that from home ^^

**Comment by aintgoin2goa** [comment]
You're still behind a proxy (were you on VPN)?

**Comment by matthew-andrews** [comment]
oh possibly

**Pull request merged**